src/main/java/com/nerodesk/om/aws/AwsDoc.java

```diff
@@ -120,6 +120,11 @@ public void read(@NotNull final OutputStream output) throws IOException {
     @Override
     public void write(final InputStream input, final long size)
         throws IOException {
+        if (size == 0) {
```

Comment:
@longtimeago from the article above
>The idea was that once completed, the implementation of a class could only be modified to correct errors; new or changed features would require that a different class be created.

I think `SmallDoc` is the new feature, a `Doc` instance can be use with or without this.
but zero length checking is not a new feature. This bug is that it not checked before...

never mind I will implement `SafeDoc`